// phys/pinball.js
/**
 * Pinball 
 * 
 */

var Box2D = require('box2dweb-commonjs');

var Pinball = function (_x, _y) {
	this.bodyDef = new Box2D.b2BodyDef;

	this.fixDef = new Box2D.b2FixtureDef;
	this.fixDef.shape = new Box2D.b2CircleShape( 20 );
	this.fixDef.restitution = .5;
	this.fixDef.density = 1;

	this.bodyDef.userData = "pinball";
	this.bodyDef.type = Box2D.b2Body.b2_dynamicBody;	
	// this.bodyDef.position.x = _x, 
	// this.bodyDef.position.x = _y; 

	world.CreateBody( this.bodyDef ).CreateFixture( this.fixDef );

	// this.pinball = world.CreateBody( this.bodyDef ).CreateFixture( this.fixDef );
	// // console.log(this.pinball);

	var listener = new Box2D.b2ContactListener;

	listener.BeginContact = function(contact) {
		var collidingBody = contact.GetFixtureB().GetBody().GetUserData();
		if(collidingBody == "Can"){
			console.log("Pinball hit the can");
		}
	}
	
	listener.EndContact = function(contact) {
	}

	listener.PostSolve = function(contact, impulse) {
		var collidingBody = contact.GetFixtureB().GetBody().GetUserData();
		
		if(collidingBody == "Can") {
			console.log(impulse);	
			// self.socket.emit('hit can', impulse);
		}
		if(collidingBody == "Flipper") {
			console.log(impulse);	
			// self.socket.emit('ball out', impulse);
		}		
	}

	listener.PreSolve = function(contact, oldManifold) {

	}

	world.SetContactListener(listener);
}

Pinball.prototype.draw = function() {
	
}

Pinball.prototype.isDead = function() {
	this.pos = this.pinball.m_body.GetPosition();

	// NEED TO FIGURE OUT HOW TO COMPUTE THE BOUNDS OF THE WORLD
	// if(this.pos.y > world.) {
	// 	return true;
	// }
	return false;
};

module.exports = Pinball;
